### PR TITLE
[type:feat] upgrade dockerfile java runtime version 8 to 17

### PR DESCRIPTION
**Pull request by tomsun28**
- upgrade dockerfile java runtime version 8 to 17, use image `openjdk:17-jdk-alpine` 

**Do not merge now, please consider merge after v2.6.1 version realease.**

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by tomsun28** [comment]
**Dubbo 2.x may not work with Java 17**
2024-03-15 18:14:56 Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @6b71769e


```
024-03-15 18:14:56 Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2024-03-15 18:14:56 2024-03-15 10:14:56.072 ERROR 7 --- [           main] o.s.boot.SpringApplication               : Application run failed
2024-03-15 18:14:56 
2024-03-15 18:14:56 org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.apache.shenyu.examples.dubbo.api.service.DubboTestService': Instantiation of bean failed; nested exception is java.lang.ExceptionInInitializerError
2024-03-15 18:14:56     at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateBean(AbstractAutowireCapableBeanFactory.java:1334) ~[spring-beans-5.3.30.jar!/:5.3.30]
2024-03-15 18:14:56     at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1232) ~[spring-beans-5.3.30.jar!/:5.3.30]
2024-03-15 18:14:56     at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:582) ~[spring-beans-5.3.30.jar!/:5.3.30]
2024-03-15 18:14:56     at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542) ~[spring-beans-5.3.30.jar!/:5.3.30]
2024-03-15 18:14:56     at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.30.jar!/:5.3.30]
2024-03-15 18:14:56     at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.30.jar!/:5.3.30]
2024-03-15 18:14:56     at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.30.jar!/:5.3.30]
2024-03-15 18:14:56     at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.30.jar!/:5.3.30]
2024-03-15 18:14:56     at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:955) ~[spring-beans-5.3.30.jar!/:5.3.30]
2024-03-15 18:14:56     at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:921) ~[spring-context-5.3.30.jar!/:5.3.30]
2024-03-15 18:14:56     at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583) ~[spring-context-5.3.30.jar!/:5.3.30]
2024-03-15 18:14:56     at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:147) ~[spring-boot-2.7.17.jar!/:2.7.17]
2024-03-15 18:14:56     at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:732) ~[spring-boot-2.7.17.jar!/:2.7.17]
2024-03-15 18:14:56     at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:409) ~[spring-boot-2.7.17.jar!/:2.7.17]
2024-03-15 18:14:56     at org.springframework.boot.SpringApplication.run(SpringApplication.java:308) ~[spring-boot-2.7.17.jar!/:2.7.17]
2024-03-15 18:14:56     at org.springframework.boot.SpringApplication.run(SpringApplication.java:1300) ~[spring-boot-2.7.17.jar!/:2.7.17]
2024-03-15 18:14:56     at org.springframework.boot.SpringApplication.run(SpringApplication.java:1289) ~[spring-boot-2.7.17.jar!/:2.7.17]
2024-03-15 18:14:56     at org.apache.shenyu.examples.alibaba.dubbo.service.TestAlibabaDubboApplication.main(TestAlibabaDubboApplication.java:37) ~[classes!/:2.7.0-SNAPSHOT]
2024-03-15 18:14:56     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
2024-03-15 18:14:56     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
2024-03-15 18:14:56     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
2024-03-15 18:14:56     at java.base/java.lang.reflect.Method.invoke(Method.java:568) ~[na:na]
2024-03-15 18:14:56     at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49) ~[shenyu-examples-alibaba-dubbo-service.jar:2.7.0-SNAPSHOT]
2024-03-15 18:14:56     at org.springframework.boot.loader.Launcher.launch(Launcher.java:108) ~[shenyu-examples-alibaba-dubbo-service.jar:2.7.0-SNAPSHOT]
2024-03-15 18:14:56     at org.springframework.boot.loader.Launcher.launch(Launcher.java:58) ~[shenyu-examples-alibaba-dubbo-service.jar:2.7.0-SNAPSHOT]
2024-03-15 18:14:56     at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65) ~[shenyu-examples-alibaba-dubbo-service.jar:2.7.0-SNAPSHOT]
2024-03-15 18:14:56 Caused by: java.lang.ExceptionInInitializerError: null
2024-03-15 18:14:56     at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:na]
2024-03-15 18:14:56     at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77) ~[na:na]
2024-03-15 18:14:56     at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:na]
2024-03-15 18:14:56     at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[na:na]
2024-03-15 18:14:56     at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480) ~[na:na]
2024-03-15 18:14:56     at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:211) ~[spring-beans-5.3.30.jar!/:5.3.30]
2024-03-15 18:14:56     at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:87) ~[spring-beans-5.3.30.jar!/:5.3.30]
2024-03-15 18:14:56     at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateBean(AbstractAutowireCapableBeanFactory.java:1326) ~[spring-beans-5.3.30.jar!/:5.3.30]
2024-03-15 18:14:56     ... 25 common frames omitted
2024-03-15 18:14:56 Caused by: java.lang.IllegalStateException: fail to create adaptive instance: java.lang.IllegalStateException: Can not create adaptive extension interface com.alibaba.dubbo.rpc.Protocol, cause: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @6b71769e
2024-03-15 18:14:56     at com.alibaba.dubbo.common.extension.ExtensionLoader.getAdaptiveExtension(ExtensionLoader.java:448) ~[dubbo-2.6.5.jar!/:2.6.5]
2024-03-15 18:14:56     at com.alibaba.dubbo.config.ServiceConfig.<clinit>(ServiceConfig.java:74) ~[dubbo-2.6.5.jar!/:2.6.5]
2024-03-15 18:14:56     ... 33 common frames omitted
2024-03-15 18:14:56 Caused by: java.lang.IllegalStateException: Can not create adaptive extension interface com.alibaba.dubbo.rpc.Protocol, cause: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @6b71769e
2024-03-15 18:14:56     at com.alibaba.dubbo.common.extension.ExtensionLoader.createAdaptiveExtension(ExtensionLoader.java:725) ~[dubbo-2.6.5.jar!/:2.6.5]
2024-03-15 18:14:56     at com.alibaba.dubbo.common.extension.ExtensionLoader.getAdaptiveExtension(ExtensionLoader.java:444) ~[dubbo-2.6.5.jar!/:2.6.5]
2024-03-15 18:14:56     ... 34 common frames omitted
2024-03-15 18:14:56 Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @6b71769e
2024-03-15 18:14:56     at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354) ~[na:na]
2024-03-15 18:14:56     at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297) ~[na:na]
2024-03-15 18:14:56     at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:199) ~[na:na]
2024-03-15 18:14:56     at java.base/java.lang.reflect.Method.setAccessible(Method.java:193) ~[na:na]
2024-03-15 18:14:56     at javassist.ClassPool.toClass2(ClassPool.java:1181) ~[javassist-3.20.0-GA.jar!/:na]
2024-03-15 18:14:56     at javassist.ClassPool.toClass(ClassPool.java:1164) ~[javassist-3.20.0-GA.jar!/:na]
2024-03-15 18:14:56     at javassist.CtClass.toClass(CtClass.java:1305) ~[javassist-3.20.0-GA.jar!/:na]
2024-03-15 18:14:56     at com.alibaba.dubbo.common.compiler.support.JavassistCompiler.doCompile(JavassistCompiler.java:122) ~[dubbo-2.6.5.jar!/:2.6.5]
2024-03-15 18:14:56     at com.alibaba.dubbo.common.compiler.support.AbstractCompiler.compile(AbstractCompiler.java:59) ~[dubbo-2.6.5.jar!/:2.6.5]
2024-03-15 18:14:56     at com.alibaba.dubbo.common.compiler.support.AdaptiveCompiler.compile(AdaptiveCompiler.java:46) ~[dubbo-2.6.5.jar!/:2.6.5]
2024-03-15 18:14:56     at com.alibaba.dubbo.common.extension.ExtensionLoader.createAdaptiveExtensionClass(ExtensionLoader.java:741) ~[dubbo-2.6.5.jar!/:2.6.5]
2024-03-15 18:14:56     at com.alibaba.dubbo.common.extension.ExtensionLoader.getAdaptiveExtensionClass(ExtensionLoader.java:734) ~[dubbo-2.6.5.jar!/:2.6.5]
2024-03-15 18:14:56     at com.alibaba.dubbo.common.extension.ExtensionLoader.createAdaptiveExtension(ExtensionLoader.java:723) ~[dubbo-2.6.5.jar!/:2.6.5]
2024-03-15 18:14:56     ... 35 common frames omitted
```

**Comment by Aias00** [comment]
still working，not finished

**Comment by tomsun28** [comment]
👍👍👍 this is the first time all checks have passed 😭😭

**Comment by Aias00** [comment]
@tomsun28 this job is done

**Comment by tomsun28** [comment]
> this job is done

@Aias00 👍👍 great!，@moremind please help to review and merge.

**Comment by moremind** [comment]
LGTM

File: shenyu-dist/shenyu-admin-dist/docker/Dockerfile [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM alpine AS prepare
+FROM centos:7 AS prepare
```
Comment (moremind) [inline shenyu-dist/shenyu-admin-dist/docker/Dockerfile:17]: alpine is 5.9M, but centos 7 is 200M, do you compare the package size?

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-dist/shenyu-admin-dist/docker/Dockerfile (hunk 1 above)
Comment (Aias00) [reply shenyu-dist/shenyu-admin-dist/docker/Dockerfile:17]: i will choose another base image later

Comment (Aias00) [reply shenyu-dist/shenyu-admin-dist/docker/Dockerfile]: @moremind  hi, i have replace the base image. refer to https://thecattlecrew.net/2022/11/07/preparing-for-spring-boot-3-choose-the-right-java-17-base-image/

**Review by yu199195 — APPROVED** [review]

**Review by yu199195 — DISMISSED** [review]

**Pull request merged**